test/staging/explicit-resource-management/disposable-stack-adopt-and-defer.js

```diff
@@ -49,7 +49,7 @@ assert.compareArray(valuesNormal, [44, 43, 42]);
 function TestDisposableStackDeferOnDisposedStack() {
   let stack = new DisposableStack();
   stack.dispose();
-  stack.defer(() => console.log(42));
+  stack.defer(() => print(42));
```

Comment:
```suggestion
  stack.defer(() => { throw new TypeError("Code should not reach here"); });
  throw new Error("Code should not reach here");
```

It's been a while I use Test262, but I believe this might resolve the issue better and confirm no run runs post .defer as well.

Reply:
Given

``` 
3. If disposableStack.[[DisposableState]] is disposed, throw a ReferenceError exception.
4. If IsCallable(onDispose) is false, throw a TypeError exception.
``` 

Line 52 can also use a simple `stack.defer('')`;` so it can observe the ReferenceError in the correct order.